### PR TITLE
[SEDONA-15] Add ST_MinimumBoundingRadius and ST_MinimumBoundingCircle functions

### PR DESCRIPTION
**Pull request by michaelmerg**
## Is this PR related to a proposed Issue?
https://issues.apache.org/jira/browse/SEDONA-15

## What changes were proposed in this PR?
Add the new geometry processing functions:
- ST_MinimumBoundingRadius
- ST_MinimumBoundingCircle

## How was this patch tested?
See unit tests

## Did this PR include necessary documentation updates?
Yes

### CODE REVIEW

File: docs/api/sql/GeoSparkSQL-Function.md [hunk 1]
```diff
@@ -583,4 +583,32 @@ FROM df
...
+Output: `POINT (2 1)`
+
+
+## ST_MinimumBoundingRadius
+
+Introduction: Returns a struct containing the center point and radius of the smallest circle that contains a geometry.
+
+Format: `ST_MinimumBoundingRadius(geom: geometry)`
+
+Since: `v1.?.?`
```
Comment (jiayuasu) [inline docs/api/sql/GeoSparkSQL-Function.md:595]: Please put Since: `1.0.1` here.

**Review by jiayuasu — CHANGES_REQUESTED** [review]
Thanks for your contribution! Can you please add some test cases for the two new functions?

You can add your test cases here: https://github.com/apache/incubator-sedona/blob/master/sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala#L852

File: docs/api/sql/GeoSparkSQL-Function.md (hunk 1 above)
Comment (michaelmerg) [reply docs/api/sql/GeoSparkSQL-Function.md]: Updated

**Comment by michaelmerg** [comment]
Sorry, forgot to commit the test cases. 
I'm not 100% happy with the `ST_MinimumBoundingCircle` test case because of the odd numbers when comparing the strings. Do you have any suggestions to improve of make it more robust?

**Comment by jiayuasu** [comment]
@michaelmerg To make it robust, since ST_ functions internally call JTS functions, so probably you can call JTS function to validate the result in the test, instead of hard-coding the test result? What do you think?

**Comment by michaelmerg** [comment]
@jiayuasu Yes, that's a good idea. I updated the test case.
Is there anything else that I need to do before we can merge this PR?

**Review by jiayuasu — APPROVED** [review]

**Comment by jiayuasu** [comment]
@michaelmerg Thank you very much for the contribution!

**Pull request merged**